apps/storybook/components/funke/IdCardPinScreen.stories.tsx

```diff
@@ -0,0 +1,243 @@
+        style={StyleSheet.absoluteFillObject}
+      />
+      <XStack justifyContent="space-between">
+        <YStack gap="$1">
+          <Paragraph secondary>Personalausweis</Paragraph>
+          <Paragraph size="$6" fontWeight="$regular">
+            {userName ?? '********'}
+          </Paragraph>
+        </YStack>
+        <Stack>
+          <Image src={issuerImage} width={75} height={75} resizeMode="contain" />
+        </Stack>
+      </XStack>
+      <XStack justifyContent="flex-end">
+        {/* TODO: background color variables */}
+        <Circle size="$3" backgroundColor={icon === 'complete' ? '#C9D6BD' : '#02010033'}>
+          {iconMapping[icon]}
+        </Circle>
+      </XStack>
+    </YStack>
+  )
+}
+
+const IdCardPinScreen = ({ pinLength, onPinComplete, state, onGoToWallet, userName }: IdCardPinScreenProps) => {
+  const { title, icon } = stateMapping[state]
+  const [pin, setPin] = useState<string[]>(new Array(pinLength).fill(''))
+  const inputRefs = useRef<Array<TextInput | null>>([])
+  const programmaticFocus = useRef(false)
+
+  const isIndexFocusable = (index: number) => {
+    const firstFocusableIndex = pin.findIndex((digit) => digit === '')
+    return firstFocusableIndex === -1 || index <= firstFocusableIndex
+  }
+
+  const setPinValue = (digit: string, index: number) => {
+    const sanitized = digit.replace(/[^0-9]/g, '')
+    if (sanitized === '') return
+
+    setPin((pin) => {
+      const newPin = [...pin]
+      newPin[index] = sanitized
+      return newPin
+    })
+
+    // Focus next input
+    if (index < pin.length - 1) {
+      programmaticFocus.current = true
+      inputRefs.current[index + 1]?.focus()
+    } else if (index === pin.length - 1) {
+      inputRefs.current[index]?.blur()
+      onPinComplete(pin.join(''))
+    }
+  }
+
+  const focusInput = (index: number) => {
+    // We programmatically focus, but the user can also focus. This is a hack to
+    // know when we are in programmatic mode
+    if (programmaticFocus.current) {
+      programmaticFocus.current = false
+      return
+    }
+
+    // If e.g. the user presses the 3rd pin, but 2nd pin is still
+    // empty it will focus the 2nd pin
+    const firstEmptyIndex = pin.findIndex((digit) => digit === '')
+    if (firstEmptyIndex !== -1 && firstEmptyIndex < index) {
+      inputRefs.current[firstEmptyIndex]?.focus()
+    } else {
+      inputRefs.current[index]?.focus()
+    }
+  }
+
+  return (
+    <Page gap="$6" justifyContent="space-between">
+      <ProgressBar value={25} />
+      <YStack flexBasis={0} flexGrow={2} flexShrink={1} justifyContent="space-between">
+        <Heading variant="title">{title}</Heading>
+        <XStack display={state === 'enterPin' ? 'flex' : 'none'} gap="$3" justifyContent="center">
+          {pin.map((digit, index) => (
+            // biome-ignore lint/suspicious/noArrayIndexKey: index is the correct key here
+            <YStack key={index} maxWidth={35} flex-1 justifyContent="center">
```

Comment:
IIRC @janrtvld this caused some very very weird errors in the idcrypt pin screen right?